incubator/hnc/docs/user-guide/quickstart.md

```diff
@@ -0,0 +1,656 @@
+cluster-wide permissions to create namespaces, but you do have a RoleBinding in
+`team-a` that gives you permission to create a `SubnamespaceAnchor` object in
+that namespace.
+
+Now you would like to create three subnamespaces for services owned by your
+team, say `service-1`, `service-2` and `service-3`:
+
+```bash
+kubectl hns create service-1 -n team-a
+kubectl hns create service-2 -n team-a
+kubectl hns create service-3 -n team-a
+kubectl hns tree team-a
+```
+
+Expected output:
+
+```
+team-a
+├── [s] service-1
+├── [s] service-2
+└── [s] service-3
+
+[s] Indicates subnamespace
+```
+
+Now if you want to add a `dev` subnamespace under `service-1`, you can:
+
+```bash
+kubectl hns create dev -n service-1
+kubectl hns tree team-a
+```
+Output:
+```
+team-a
+├── [s] service-1
+│    └── [s] dev
+├── [s] service-2
+└── [s] service-3
+```
+
+As with regular namespaces ("full namespaces," in HNC terms), subnamespaces must
+have unique names. For example, you cannot add a second `dev` subnamespace under
+`service-2`. The webhook will prevent you from creating that subnamespace:
+
+```bash
+kubectl hns create dev -n service-2
+# Error: The requested namespace dev already exists. Please use a different name.
+```
+
+You can also delete subnamespaces by deleting the anchor in the parent
+namespace. For example:
+
+```bash
+kubectl delete ns service-3
+# doesn't work
+
+kubectl delete subns service-3 -n team-a
+# service-3 is deleted
+```
+
+Note that `subns` is a short form for `subnamespaceanchor` or
+`subnamespaceanchor.hnc.x-k8s.io`.
+
+
+Now try to delete `service-1` in the same way, but you'll see it doesn't work:
+
+```bash
+kubectl delete subns service-1 -n team-a
+# forbidden
+```
+
+The reason for this is that `service-1` contains its own subnamespace that would
+be deleted with it, because deleting a namespace also deletes all the anchors in
+that namespace, and therefore all its subnamespaces. This is very dangerous!
+
+Therefore, just like rm -r won't recursively delete a tree of directories
+without an -f option, HNC prevents you from deleting any namespace that contains
+subnamespaces by default.
+
+HNC will only allow recursive deletion of subnamespaces if those subnamespace,
+or any of their ancestors, have cascading deletion explicitly set. For example,
+to delete `service-1`, we first enable cascading deletion and then remove it:
+
+```bash
+# HNC v0.6:
+kubectl hns set service-1 --allowCascadingDeletion
+
+# HNC v0.5:
+kubectl hns set service-1 --allowCascadingDelete
+
+# Short form (both v0.5 and v0.6):
+kubectl hns set service-1 -a
+```
+
+Now you can (unsafely!) delete `service-1`:
+
+```bash
+kubectl delete subns service-1 -n team-a
+```
+
+There's an important difference between subnamespaces and regular child
+namespace, also known as a full namespace. A subnamespace is created by HNC
+based from an anchor in the parent; when that anchor is deleted, the
```

Comment:
ditto: based on?

Reply:
Changed to "due to an anchor being created in the parent"